pkg/apis/backups/v1alpha1/util.go

```diff
@@ -0,0 +1,127 @@
+/*
+Copyright Kurator Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Destination defines a fleet or specific clusters.
+type Destination struct {
+	// Fleet is the name of fleet.
+	// The field, in combination with ClusterSelector, can determine a set of clusters.
+	// In addition to this approach, users can also directly specify clusters through the field Clusters.
+	// +optional
+	Fleet string `json:"fleet,omitempty"`
+
+	// ClusterSelector specifies the selectors to select the clusters within the fleet.
+	// If unspecified, all clusters in the fleet will be selected.
+	// The field will only take effect when Fleet is set.
+	// +optional
+	ClusterSelector *ClusterSelector `json:"clusterSelector,omitempty"`
+
+	// Clusters determine a set of clusters as destination clusters.
+	// The field will only take effect when Fleet is not set.
+	// +optional
+	Clusters []*corev1.ObjectReference `json:"clusters,omitempty"`
```

Comment:
so this means clusters will have lower priority

Reply:
yes.

Alternatively, we could allow both to coexist and then take the union.

Reply:
We can simplify the design by removing the `ClusterSelector`. If both `Fleet` and `Clusters` are specified, we will take their union as the final set of clusters.